xds/src/test/java/io/grpc/xds/ServerXdsClientTest.java

```diff
@@ -639,6 +636,8 @@ public void streamClosedAndRetry() {
     verify(requestObserver)
         .onNext(eq(buildDiscoveryRequest(getNodeToVerify(), "",
             ResourceType.LDS.typeUrlV2(), "")));
+    // expect only 1 request: for LDS
+    verify(requestObserver, times(1)).onNext(any(DiscoveryRequest.class));
```

Comment:
Isn't this just what the previous line verified, except it adds the `times(1)` restriction? I think all you need is `verifyNoMoreInteractions(requestObserver)` at the end of the lifecycle of each `requestObserver` instance (note there are several `requestObserver` updates in this test as more than one RPCs are made).

Reply:
So it does verify that there is only request which is the previous one (for LDS) and no more. So `verifyNoMoreInteractions(requestObserver)` also works but  in terms of qualitative testing there is no difference.